IPython/terminal/embed.py

```diff
@@ -130,6 +147,10 @@ def __call__(self, header='', local_ns=None, module=None, dummy=None,
         if self.exit_msg is not None:
             print(self.exit_msg)
 
+        if self.should_raise:
+            raise KillEmbeded('This instance has been marked as must raise on exit.')
```

Comment:
This error message is not very clear. Let's say something like "Embedded IPython raising error, as user requested"